src/pages/patientView/structuralVairnt/PatientViewStructuralVariantTable.tsx

```diff
@@ -0,0 +1,403 @@
+import * as React from 'react';
+import { computed } from 'mobx';
+import { observer } from 'mobx-react';
+import { PatientViewPageStore } from '../clinicalInformation/PatientViewPageStore';
+import LazyMobXTable, {
+    Column,
+} from 'shared/components/lazyMobXTable/LazyMobXTable';
+import { StructuralVariant } from 'cbioportal-ts-api-client';
+import TumorColumnFormatter from '../mutation/column/TumorColumnFormatter';
+import HeaderIconMenu from '../mutation/HeaderIconMenu';
+import GeneFilterMenu from '../mutation/GeneFilterMenu';
+import PanelColumnFormatter from 'shared/components/mutationTable/column/PanelColumnFormatter';
+import * as _ from 'lodash';
+import { MakeMobxView } from 'shared/components/MobxView';
+import LoadingIndicator from 'shared/components/loadingIndicator/LoadingIndicator';
+import ErrorMessage from 'shared/components/ErrorMessage';
+import AnnotationColumnFormatter from './column/AnnotationColumnFormatter';
+import AppConfig from 'appConfig';
+import { ServerConfigHelpers } from 'config/config';
+import ChromosomeColumnFormatter from 'shared/components/mutationTable/column/ChromosomeColumnFormatter';
+
+export interface IPatientViewStructuralVariantTableProps {
+    store: PatientViewPageStore;
+    onSelectGenePanel?: (name: string) => void;
+}
+
+type CNATableColumn = Column<StructuralVariant[]> & { order: number };
+
+class StructuralVariantTableComponent extends LazyMobXTable<
+    StructuralVariant[]
+> {}
+
+@observer
+export default class PatientViewStructuralVariantTable extends React.Component<
+    IPatientViewStructuralVariantTableProps,
+    {}
+> {
+    constructor(props: IPatientViewStructuralVariantTableProps) {
+        super(props);
+    }
+
+    @computed get columns() {
+        const columns: CNATableColumn[] = [];
+        const numSamples = this.props.store.sampleIds.length;
+
+        if (numSamples >= 2) {
+            columns.push({
+                name: 'Samples',
+                render: (d: StructuralVariant[]) =>
+                    TumorColumnFormatter.renderFunction(
+                        d.map(datum => ({
+                            sampleId: datum.sampleId,
+                            entrezGeneId: datum.site1EntrezGeneId,
+                        })),
+                        this.props.store.sampleManager.result!,
+                        this.props.store.sampleToStructuralVariantGenePanelId
```

Comment:
just want to make sure this is awaited somewhere.

Reply:
yes as a matter of fact i think its better for `columns` to be a `remoteData` and then just `await` it down in `tableUI` before using it

Reply:
Yes, it is awaited at other place. I was following the same model as other tables in patient page. Yeah, even I prefer to be `remoteData`. I'll update it